modules/compliance-tests/src/smithy4s/compliancetests/internals/ClientHttpComplianceTestCase.scala

```diff
@@ -91,44 +91,45 @@ private[compliancetests] class ClientHttpComplianceTestCase[
       testCase.method.toLowerCase(),
       request.method.name.toLowerCase()
     )
-    val ioAsserts = bodyAssert +:
+    val ioAsserts: List[F[ComplianceResult]] = bodyAssert +:
       List(
         assert.testCase.checkHeaders(testCase, request.headers),
         uriAssert,
         methodAssert
       )
-        .map(_.pure[IO])
+        .map(_.pure[F])
     ioAsserts.combineAll
   }
 
   private[compliancetests] def clientRequestTest[I, E, O, SE, SO](
       endpoint: service.Endpoint[I, E, O, SE, SO],
       testCase: HttpRequestTestCase
-  ): ComplianceTest[IO] = {
-    type R[I_, E_, O_, SE_, SO_] = IO[O_]
+  ): ComplianceTest[F] = {
+    type R[I_, E_, O_, SE_, SO_] = F[O_]
 
     val inputFromDocument = Document.Decoder.fromSchema(endpoint.input)
-    ComplianceTest[IO](
+    ComplianceTest[F](
       name = endpoint.id.toString + "(client|request): " + testCase.id,
       run = {
         val input = inputFromDocument
           .decode(testCase.params.getOrElse(Document.obj()))
-          .liftTo[IO]
+          .liftTo[F]
 
-        deferred[Request[IO]].flatMap { requestDeferred =>
+        deferred[Request[F]].flatMap { requestDeferred =>
           val app = HttpRoutes
-            .of[IO] { case req =>
+            .of[F] { case req =>
               req.body.compile.toVector
                 .map(fs2.Stream.emits(_))
                 .map(req.withBodyStream(_))
                 .flatMap(requestDeferred.complete(_))
-                .as(Response[IO]())
+                .as(Response[F]())
             }
             .orNotFound
 
```

Comment:
`HttpRoutes.of{...}.orNotFound` is `HttpApp.apply`, or even just plain `Kleisli`

Reply:
That's not exactly accurate, but the point remains that you should indeed use `HttpApp.apply` in this instance

Reply:
> That's not exactly accurate

Care to elaborate? I meant in this particular instance (the function passed to `HttpRoutes.of` is total).

Reply:
>  I meant in this particular instance

That addition to your initial statement makes it accurate 😄